### PR TITLE
db-3187 [next-wordpress] Add Authenticated API to our demo

### PR DESCRIPTION
**Pull request by lotusbaba**
## What changes were made?
Added page - examples/auth-api.jsx

## Where were the changes made?
 
 
next-wordpress

## How have the changes been tested?
Yes via localhost

## Additional information

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

File: starters/next-wordpress-starter/pages/examples/auth-api.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            <>
+              <p>
+                Next.js was unable to make an authorized request to the
+                Wordpress API. Please check your .env.development.local file to
+                ensure that your WP_APPLICATION_USERNAME and
+                WP_APPLICATION_PASSWORD are set correctly.
+              </p>
+              <p>
+                For more information on how to set these values, please see{" "}
+                <a href="https://github.com/pantheon-systems/decoupled-kit-js/blob/canary/web/docs/Frontend%20Starters/Next%20Wordpress/setting-environment-variables.md">
```
Comment (CobyPear) [inline starters/next-wordpress-starter/pages/examples/auth-api.jsx:39]:
```suggestion
                <a href="https://github.com/pantheon-systems/decoupled-kit-js/blob/canary/web/docs/Frontend%20Starters/Next.js/Next.js%20%2B%20WordPress/setting-environment-variables.md">
```
This link needs to be updated

File: starters/next-wordpress-starter/pages/examples/auth-api.jsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        <Link passHref href="/">
+          <span className="w-full underline cursor-pointer">Home &rarr;</span>
+        </Link>
+
+        <div className="mt-12 max-w-lg mx-auto lg:grid-cols-3 lg:max-w-screen-lg">
+          {SSRPosts?.length > 0 ? (
+            <p>
+              🎉 Next.js was able to successfully make an authenticated request
+              to Wordpress! 🎉
```
Comment (CobyPear) [inline starters/next-wordpress-starter/pages/examples/auth-api.jsx:27]:
```suggestion
              to WordPress! 🎉
```
WP gets mad when you don't capitalize the "P"

File: starters/next-wordpress-starter/pages/examples/auth-api.jsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            </p>
+          ) : (
+            <>
+              <p>
+                Next.js was unable to make an authorized request to the
+                Wordpress API. Please check your .env.development.local file to
+                ensure that your WP_APPLICATION_USERNAME and
+                WP_APPLICATION_PASSWORD are set correctly.
```
Comment (CobyPear) [inline starters/next-wordpress-starter/pages/examples/auth-api.jsx:35]:
```suggestion
                ensure that your <code>WP_APPLICATION_USERNAME</code> and
                <code>WP_APPLICATION_PASSWORD</code> are set correctly.
```
Wrap these in code tags to match the formatting of the `next-drupal-starter`

File: starters/next-wordpress-starter/pages/examples/auth-api.jsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+          {SSRPosts?.length > 0 ? (
+            <p>
+              🎉 Next.js was able to successfully make an authenticated request
+              to Wordpress! 🎉
+            </p>
+          ) : (
+            <>
+              <p>
+                Next.js was unable to make an authorized request to the
+                Wordpress API. Please check your .env.development.local file to
```
Comment (CobyPear) [inline starters/next-wordpress-starter/pages/examples/auth-api.jsx:33]:
```suggestion
                WordPress API. Please check your .env.development.local file to
```

File: starters/next-wordpress-starter/pages/examples/auth-api.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+              node {
+                altText
+                sourceUrl
+              }
+            }
+          }
+        }
+      }
+    }
+  `;
```
Comment (CobyPear) [inline starters/next-wordpress-starter/pages/examples/auth-api.jsx:77]: All we really need here is the `id`

File: starters/next-wordpress-starter/pages/examples/auth-api.jsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+  if (process.env.DEBUG_MODE) {
+    console.log(process.env.WP_APPLICATION_USERNAME);
+    console.log(
+      btoa(
+        `${process.env.WP_APPLICATION_USERNAME}:${process.env.WP_APPLICATION_PASSWORD}`
+      )
+    );
+    console.log(edges);
+  }
```
Comment (CobyPear) [inline starters/next-wordpress-starter/pages/examples/auth-api.jsx:94]: Let's remove this

File: starters/next-wordpress-starter/pages/examples/auth-api.jsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+      }
+    }
+  `;
+
+  const {
+    posts: { edges },
+  } = await client.request(query);
+
+  const menuItems = await getFooterMenu();
+  const SSRPosts = edges.map(({ node }) => node);
```
Comment (CobyPear) [inline starters/next-wordpress-starter/pages/examples/auth-api.jsx:84]: we can be more specific here, maybe call this variable `privatePosts`

**Review by CobyPear — CHANGES_REQUESTED** [review]
Some comments mostly re: naming and formatting

File: starters/next-wordpress-starter/pages/examples/auth-api.jsx (hunk 1 above)
Comment (CobyPear) [reply starters/next-wordpress-starter/pages/examples/auth-api.jsx]: @lotusbaba this link is still incorrect. See the suggested change

**Review by CobyPear — CHANGES_REQUESTED** [review]
The link to the doc still seems incorrect, but looks good otherwise!

**Comment by lotusbaba** [comment]
@CobyPear Resolved your last comment

**Review by CobyPear — APPROVED** [review]
:shipit:

**Pull request merged**